### PR TITLE
Run benchmark tests on CI

### LINKED ISSUE #8: Add running the unit tests to CI
Important: the unit tests should be run by CI, in particular the benchmarking unit test, which ensures that the algorithm wasn't modified and still produces the same output for the same input.

### PR DESCRIPTION
**Pull request by kryksyh**
Fixes: #8

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Build and Run Benchmark
+
+on:
+  push:
+    branches: [master]
```
Comment (saintmatthieu) [inline .github/workflows/benchmark.yml]: In wonder, should this run on dev branches, too ? If I touch the algorithm this early feedback could be useful.

File: tests/benchmarking/TatumQuantizationFitBenchmarking.cpp [hunk 2]
```diff
@@ -144,8 +144,9 @@ TEST_CASE("TatumQuantizationFitBenchmarking")
    // Verify that the input has not changed between this run and the previous
    // one, or peformance of the algorithm can't be compared either.
    constexpr auto previousChecksum = -177205.328125f;
+   // TODO: FIX THE TOLERANCE
    REQUIRE(
-      ValueIsUnchanged(outputDir / "checksum.txt", previousChecksum, checksum));
+      ValueIsUnchanged(outputDir / "checksum.txt", previousChecksum, checksum, 1.0f));
```
Comment (saintmatthieu) [inline tests/benchmarking/TatumQuantizationFitBenchmarking.cpp:150]: 1.0f, so much ? At the scale of 177205 that's probably negligible, yeah, ok.  How about platform-dependent references ? Unreasonably much work ?
What TODO do you have in mind?

**Review by saintmatthieu — APPROVED** [review]

**Comment by saintmatthieu** [comment]
Oh, it fails on Linux ?

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (kryksyh) [reply .github/workflows/benchmark.yml]: make sense

File: tests/benchmarking/TatumQuantizationFitBenchmarking.cpp (hunk 2 above)
Comment (kryksyh) [reply tests/benchmarking/TatumQuantizationFitBenchmarking.cpp:150]: > How about platform-dependent references 

I need first to understand what this checksum is :)
Is it possible not to use FP at all and use a more stable checksum instead? Can you give a high-level overview how it works currently?

Comment (saintmatthieu) [reply tests/benchmarking/TatumQuantizationFitBenchmarking.cpp:150]: It's to make sure that the set of input files hasn't changed before we do the benchmarking, and that the same evaluation is indeed being reproduced.
I'll talk about it tomorrow in the talk :D
> Is it possible not to use FP at all and use a more stable checksum instead?

I'm open, but for now I think that will do.

**Comment by saintmatthieu** [comment]
Thanks a mil, @kryksyh, on more step towards having this repo in a proper state !

**Pull request merged**